### ISSUE TITLE
difference between LSP mode and "linter" mode; `from a import b` vs `import a.b`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5003]
### Discussed in https://github.com/microsoft/pyright/discussions/5000

<div type='discussions-op-text'>

<sup>Originally posted by **dkuettel** April 24, 2023</sup>
Hello,

I know that tensorflow doesnt have terribly good type annotations yet. However, I seem to have some jumpy behavior with the LSP and it doesnt always seem to be the same as when I run it in "batch mode" using `pyright some_file.py`:

Generally, when in LSP mode, this doesnt work:

```python
from tensorflow import keras
keras.layers.Input  # None of this is completed
```

Yet, when I do:

```python
import tensorflow as tf
tf.keras.layers.Input  # everything gets completed beautifully
```

On top of that, the first version is shown as an error in LSP mode, but when I run it using `pyright some_file.py` I get no errors.

It's not completely clear that my LSP mode and "batch mode" are running in the exact same settings, but they do use the same `pyproject.toml`.

But as for the difference in LSP mode: Does the LSP mode have some kind of timeout how deep it scans into a library to inspect things? Is that maybe why it gets it only in the second case?

cheers, Daniel.
</div>

**Comment by erictraut** [comment]
This will be fixed in the next release.

The problem is related to the (somewhat unusual) way that the `keras` module is importing one of its submodules. The keras module `keras/api/_v2/__init__.py` contains the import statement `from keras.api._v2 import keras`. This statement is targeting itself (since `keras.api._v2` refers to the file `keras/api/_v2/__init__.py`). Normally, libraries would use the relative import `from . import keras` in this situation. Pyright contained some special-case code to deal with this situation, but it didn't handle the absolute import case correctly.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.306, which I just published. It will also be included in a future release of pylance.
